.github/workflows/ci.yml

```diff
@@ -1,42 +1,50 @@
 name: CI
 on:
-  push:
-    branches:
-      - master
-    tags: '*'
   pull_request:
+  push:
+    branches: [main]
+    tags: ['*']
 jobs:
   test:
-    name: Julia ${{ matrix.version }} - ${{ matrix.os }} - ${{ matrix.arch }}
+    name: Julia ${{ matrix.version }} - ${{ matrix.os }} - ${{ matrix.arch }} - ${{ github.event_name }}
     runs-on: ${{ matrix.os }}
-    continue-on-error: ${{ matrix.version == 'nightly' }}
     strategy:
+      fail-fast: false
       matrix:
         version:
           - '1.6'
-          - '1'
-          - 'nightly'
+          - '1' # automatically expands to the latest stable 1.x release of Julia
         os:
           - ubuntu-latest
-          - macOS-latest
-          - windows-latest
         arch:
           - x64
-        exclude:
-          # Remove some configurations from the build matrix to reduce CI time.
-          # See https://github.com/marketplace/actions/setup-julia-environment
-          # Don't test on all versions
-          - {os: 'macOS-latest', version: '1.6'}
-          - {os: 'macOS-latest', version: 'nightly'}
-          - {os: 'windows-latest', version: '1.6'}
-          - {os: 'windows-latest', version: 'nightly'}
+        include:
+          - os: windows-latest
+            version: '1'
+            arch: x86
     steps:
-      - uses: actions/checkout@v1
-      - uses: julia-actions/setup-julia@latest
+      - uses: actions/checkout@v2
+      - uses: julia-actions/setup-julia@v1
         with:
           version: ${{ matrix.version }}
           arch: ${{ matrix.arch }}
-      - uses: julia-actions/julia-buildpkg@latest
-      - uses: julia-actions/julia-runtest@latest
+      - uses: julia-actions/cache@v1
+      - uses: julia-actions/julia-buildpkg@v1
+      - uses: julia-actions/julia-runtest@v1
+        env:
+          JULIA_NUM_THREADS: 4
+      - uses: julia-actions/julia-processcoverage@v1
+      - uses: codecov/codecov-action@v1
         with:
```

Comment:
I'd say that it's fine to remove code coverage again. Ensure to pass `coverage: false` to `julia-actions/julia-runtests` since this will speed up the tests (often dramatically).